homeassistant/components/compit/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""The Compit integration."""
+
+from compit_inext_api import (
+    CannotConnect,
+    CompitAPI,
+    DeviceDefinitionsLoader,
+    InvalidAuth,
+    SystemInfo,
+)
+
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import (
+    ConfigEntryAuthFailed,
+    ConfigEntryError,
+    ConfigEntryNotReady,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .coordinator import CompitConfigEntry, CompitDataUpdateCoordinator
+
+PLATFORMS = [
+    Platform.CLIMATE,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: CompitConfigEntry) -> bool:
+    """Set up Compit from a config entry."""
+
+    session = async_get_clientsession(hass)
+    api = CompitAPI(entry.data["email"], entry.data["password"], session)
+    try:
+        system_info = await api.authenticate()
+    except CannotConnect as e:
+        raise ConfigEntryNotReady(f"Error while connecting to Compit: {e}") from e
+    except InvalidAuth as e:
+        raise ConfigEntryAuthFailed(
+            f"Invalid credentials for {entry.data['email']}"
+        ) from e
+
+    if isinstance(system_info, SystemInfo):
```

Comment:
would system info ever not be of this type?

Reply:
Yes - when api return unexpected value which cannot be parsed into SystemInfo